### ISSUE TITLE
Add Mutation Diagram on Study View for Enhanced Mutation Analysis

### ISSUE THREAD

**Issue by inodb** [cBioPortal/GSoC#87]
**Background**: 
The Study View (e.g. [MSK-IMPACT 2017 study](https://www.cbioportal.org/study?id=msk_impact_2017)) of cBioPortal provides a powerful interactive way for exploring clinical and genomic data in a study (cohort of samples). 

![image](https://user-images.githubusercontent.com/840895/53684831-741e6280-3ce0-11e9-9070-eb0076449b63.png)

**Goal**: 
The overall goal of this project to enhance its capability for analyzing mutations. Specifically:
- Visualizing mutations in a gene with [mutation diagram](http://bit.ly/2tKqafv)
  - Each gene in the `Mutated Genes` table can have a submenu or icons that can lead to adding a mutation diagram
![image](https://user-images.githubusercontent.com/840895/53684805-29044f80-3ce0-11e9-8464-6f2259dc32cc.png)
- The added mutation diagram should be interactive with other charts
  - When the user selects samples with other charts, the mutation diagram will only show mutations for the selected samples
  - When selecting one or more mutations in the mutation diagram, the other charts will also reflect based on the selected samples.
  - Allow user to select a range
  - Allow zoom in

**Approach**: 
The REACT component that is used in the Mutations tab should be reused here. Enhancement will be needed, e.g. zooming capability.

**Need skills**: 
Javascript, REACT

**Possible mentors**: 
@Luke-Sikina @kalletlak @dippindots

**Comment by Kanha-13** [comment]
I wanted to contribute in this project can anyone help me with this?

**Comment by tanishq67** [comment]
Hi, I am Tanishq Gupta, a 3rd-year UnderGraduate pursuing Biological Sciences and Bio-Engineering from IIT Kanpur. I completed GSOC’21 under Open Bioinformatics Foundation last year. I have been programming since, the last 2 years, and I am very much passionate about working on projects that are related to biology.
As a web-head, I with my team created this https://www.ppociitk.in/ website in react for our PPOC club from scratch.  

I think I have the relevant skills to work on this project. 
Here, is my Resume for reference https://drive.google.com/file/d/1sMQU0juB2xShgKLDDQ8lqcHlJuSemvvO/view?usp=sharing
I am really interested in contributing to this project for this GSoC’22. Could you please let me know if there is any sub-issue that I can first fix and put up a PR on to show my competence, or shall I start working on the proposal?

PS:- The given link [cBioPortal](https://github.com/cBioPortal/GSoC/issues/www.cbioportal.org) is not working.

**Comment by tanishq67** [comment]
@Luke-Sikina @dippindots

**Comment by tanishq67** [comment]
@inodb is this issue still up for GSoC? Or is there any other mailing list that I am supposed to join? please let me know.
Thank You :)

**Comment by ao508** [comment]
hi @tanishq67, thank you for your interest in this project and apologies for the delayed response! This project is still available and I know for certain Luke is still interested in being a mentor. If you have any questions please let us know.

**Comment by tanishq67** [comment]
@ao508 sure, I am on it, will give you update in around a week :)

**Comment by MoatazNasr** [comment]
Hey is there any issues to work on before submitting the proposal ?

**Comment by AbhishekJamhoriya** [comment]
Hi @ao508,
I'm in my 3rd year at IIT Jodhpur INDIA and currently pursuing my B.tech  degree in Computer Science and Technology. I wanna apply for Gsoc and feels that this project will be a great fit for me.  I have a great knowledge of Javascript, HTML, CSS  and MERN  stack. I have done an internship where I was working on a similar thing.so if you give me a green signal I can work on the proposal part.

**Comment by ao508** [comment]
@AbhishekJamhoriya @MoatazNasr Thank you for your interest! Please feel free to begin working on the proposal and let us know if you have any additional questions. 

At this time I don't believe there are any issues to work on related to the Mutation Diagram before submitting the proposal.

**Comment by aakilkhanemon** [comment]
Hi there! Is this issue still up for GSoC2022? I am interested in contributing to this project. Could anyone please let me know if there is any chance to get involved specific project? 
Thanks for your time.

**Comment by MayaSatishRao** [comment]
Is this issue still open? I wanted to contribute and wanted to know if this is available for GSOC 2023?

**Comment by inodb** [comment]
@MayaSatishRao yes still open - note that GSoC 2023 participating organizations haven't been announced yet

**Comment by Aaryan-Musk** [comment]
@inodb  Hello there,
I am a second year student in computer science department. i want to contribute in this project now(2023). I have knowledge in react and javascript. I personally am really passionate to solve issues that can help cancer worriers. 
If i could get an email it would be extremely helpful. i really want to discuss this idea with you.

**Comment by harsh2929** [comment]
@inodb I would love to contribute in this issue, I'm a second year student doing my bachelors  in CSE from NIT-K.

**Comment by Amritanshu02** [comment]
Hello everyone,
I am Amritanshu, a CSE 3rd year student. I am currently pursuing my B.tech from Babu Banarasi Das Institue of Technology and Management, India.
I am very interested in website building and had build a number of websites like Netflix clone, staybnb clone and many more. Recently I had also developed an interest in Open Source and I would like to contribute for the organizatons in GSoC and show case my skills. I had also participated in Hacktoberfest event and was able to make successfully 4 PRs.
After carefully going through your GSoC 2023 Ideas, I had developed a keen interest for "Add Mutation Diagram on Study View for Enhanced Mutation Analysis". I being skilled in Java, HTML, CSS, JavaScript and React and also being familiar with MySQL, DBMS, Git and GitHub, believe that I will be able to make some meaningful contributions to your organizations.
I will be looking forward to become a part of your organization and show case my development skills. Also I am new to this whole GSoC so I will be looking for your guidance @Luke-Sikina @dippindots.
Thank you all.

GitHub: https://github.com/Amritanshu02
LinkedIn: https://www.linkedin.com/in/amritanshu-gupta-982529201/

**Comment by neeraj542** [comment]
Hello Everyone,
My name is Neeraj Meena, and I'm a first-year student at NIT Bhopal. I'm excited to dive into the open-source community and contribute to meaningful projects.

As a web developer, I have experience working with the MERN stack and possess proficiency in JavaScript, C++, HTML, and CSS. Currently, I'm building websites and working with ReactJS. I'm excited about the opportunity to work on a real-life project like this and contribute to the development of cBioPortal.

I'm passionate about programming and eager to apply my skills to real-world problems.

I'm interested in contributing to this project and I would like to take on issue #87. I think interactive data visualization would be a great addition to the portal, and I have experience with JavaScript. Can you assign this issue to me, please?

In particular, I am hoping to work with mentors such as @Luke-Sikina @dippindots. I am impressed by your contributions to the open-source community and would be honored to work alongside you.

GitHub: https://github.com/neeraj542
LinkedIn: https://www.linkedin.com/in/kumar-neeraj-2019/

Thank you for your time and consideration. I look forward to hearing from you soon.
